File: 11-container-with-most-water/container-with-most-water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int left = 0;
        int right = height.Length - 1;
        int maxArea = 0;

        while (left < right) {
            int minHeight = Math.Min(height[left], height[right]);
            int width = right - left;
            int currentArea = minHeight * width;
            maxArea = Math.Max(maxArea, currentArea);

            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return maxArea;
    }
}

File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let left=0;
    let right=height.length-1;
    let maxArea=0;

    while (left<right){
        let width=right-left;
        let minHeight=Math.min(height[left], height[right]);

        maxArea=Math.max(maxArea, width*minHeight);

        if(height[left]<height[right]){left++;}
        else{right--;}
    }return maxArea;
};